drivers/power_domain/power_domain_tisci.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2025 Texas Instruments
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/pm/device_runtime.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/firmware/tisci/tisci.h>
+#include <zephyr/drivers/power/ti,sci_pm_domain.h>
+LOG_MODULE_REGISTER(tisci_pd);
+
+#define DT_DRV_COMPAT ti_sci_pm_domain
+
+const struct device *dmsc;
+
+struct power_domain {
+	uint32_t devid;
+	uint32_t mode;
+};
+
+static int tisci_power_domain_on(struct power_domain *pd)
+{
+	int ret;
+
+	if (pd->mode) {
+		ret = tisci_cmd_get_device_exclusive(dmsc, pd->devid);
+	} else {
+		ret = tisci_cmd_get_device(dmsc, pd->devid);
+	}
+
+	if (ret) {
+		LOG_ERR("TISCI PM: get_device(%u) failed (%d)\n", pd->devid, ret);
+	}
+
+	return ret;
+}
+
+static int tisci_power_domain_off(struct power_domain *pd)
+{
+	int ret = tisci_cmd_put_device(dmsc, pd->devid);
+
+	if (ret) {
+		LOG_ERR("TISCI PM: put_device(%u) failed (%d)\n", pd->devid, ret);
+	}
+
+	return ret;
+}
+
+static int tisci_pd_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	struct power_domain *data = dev->data;
+
+	LOG_DBG("TISCI PM action %d on devid %d, mode %d", action, data->devid, data->mode);
+	int ret;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		return 0;
+	case PM_DEVICE_ACTION_SUSPEND:
+		return 0;
+	case PM_DEVICE_ACTION_TURN_ON:
+		ret = tisci_power_domain_on(data);
+		return ret;
+	case PM_DEVICE_ACTION_TURN_OFF:
+		ret = tisci_power_domain_off(data);
+		return ret;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int tisci_pd_init(const struct device *dev)
+{
+	dmsc = DEVICE_DT_GET(DT_NODELABEL(dmsc));
+	if (dmsc == NULL) {
+		LOG_ERR("DMSC device not found");
+		return -ENODEV;
+	}
```

Comment:
This does not work, DEVICE_DT_GET() will expand to a token like `__device_dt_ord_20` regardless of whether `dmsc` exists and has a driver enabled. This can be moved to `const struct device *dmsc = DEVICE_DT_GET(DT_NODELABEL(dmsc));` at line 17, or if you want this behavior, DEVICE_DT_GET_OR_NULL() can be used, which expands to NULL if the node dmsc has status = "disabled";